src/averageAges.js

```diff
@@ -37,7 +43,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => {
+    return withChildren !== undefined
+      ? person.sex === 'f' && people.some(item => item.mother === person.name)
+      : person.sex === 'f';
+  });
+
+  const totalAge = women.reduce((sum, item) => {
+    return sum + (item.died - item.born);
+  }, 0);
+
+  return +(totalAge / women.length).toFixed(2);
```

Comment:
Same as above, replace `+(totalAge / women.length).toFixed(2);` with `Number((totalAge / women.length).toFixed(2));` for better readability.